tests/functional/idempotency/test_idempotency.py

```diff
@@ -502,6 +478,9 @@ def test_idempotent_lambda_expired_during_request(
 
     stubber.activate()
 
+    submethod_mocked = mocker.patch.object(DynamoDBPersistenceLayer, "boto3_supports_condition_check_failure")
```

Comment:
```suggestion
    submethod_mocked = mocker.patch.object(persistence_store, "boto3_supports_condition_check_failure")
```

I was curious about this change, could you explain the effect of this to me? Out of curiosity, I ran some local tests with the True and False values and the test passed normally.

Reply:
The idea with the test was that when testing locally, you will probably have a version of the boto3 >= 1.26.164 installed.  So unless we have boto3_supports_condition_check_failure return false, the path for what happens when boto3 <  1.26.164 will not be tested locally.